pkg/network/usm/monitor.go

```diff
@@ -181,30 +180,45 @@ func (m *Monitor) Start() error {
 		return nil
 	}
 
-	var err error
+	var (
+		err error
+
+		// This value is here so that both the new way of handling protocols and
+		// the old one (used here by kafka & http2 monitoring) can coexist in
+		// this function. It SHOULD be removed once every protocol has been
+		// refactored to use the new way.
+		enabledCount int = 0
+	)
 
 	defer func() {
 		if err != nil {
 			if errors.Is(err, syscall.ENOMEM) {
 				err = fmt.Errorf("could not enable http monitoring: not enough memory to attach http ebpf socket filter. please consider raising the limit via sysctl -w net.core.optmem_max=<LIMIT>")
 			}
 
+			// Cleanup every remaining protocols
+			for _, protocol := range m.protocols {
+				protocol.PreStop(m.ebpfProgram.Manager.Manager)
+				protocol.PostStop(m.ebpfProgram.Manager.Manager)
```

Comment:
seeing that together makes me wonder if there is a good reason to have "pre-start" and "prestop"
currently the answer is no